src/backend/gl/src/lib.rs

```diff
@@ -39,6 +39,25 @@ mod window;
 #[cfg(feature = "glutin")]
 pub use window::glutin::{config_context, Headless, Surface, Swapchain};
 
+pub(crate) struct GlContainer {
+    context: gl::Gl,
+}
+
+impl GlContainer {
+    fn make_current(&self) {
+        // Unimplemented
+    }
+}
+
+impl Deref for GlContainer {
+    type Target = gl::Gl;
+    fn deref(&self) -> &gl::Gl {
+        #[cfg(feature = "glutin")]
+        self.make_current();
```

Comment:
Calling `make_current` on each context access is worrisome.
Couldn't we just switch the active contexts before and after presenting under the constraint that everything must happen in the 'main' thread?

Reply:
Most of the time calling `make_current` won't do anything, if the context is already current. Currently we call `is_current` to check, but if that proves too costly, we could make a thread_local uint containing the id (assigned by us) of the currently active context.

Switching contexts might prove useful in the future for other things, like emulating FBOs in OpenGL 2.1 (I'm not really sure how this would work out yet, so that's a future me problem). In such case, switching from and back to the `main` context might not end up being the best thing (or it might, who knows).

Until switching contexts proves too costly, I'd like to avoid adding additional constraints.